back/backend/handlers/admin/show_users.py

```diff
@@ -0,0 +1,89 @@
+from json import loads
+from flask import Blueprint, jsonify, request
+from google.cloud import ndb
+
+from backend.models.wca.person import Person
+from backend.lib import auth
+from backend.models.user import User, Roles
+
+bp = Blueprint('show_users', __name__)
+client = ndb.Client()
+
+
+@bp.route('/get_users')
+def get_users():
+    with client.context():
+        me = auth.user()
+        if not me or not me.HasAnyRole(Roles.AdminRoles()):
+            return jsonify({"error": "Forbidden"}), 403
+
+        # Pagination
+        page = request.args.get('page', 1, type=int)
+        per_page = request.args.get('per_page', 30, type=int)
+        cursor = request.args.get('cursor', None, type=str)
+        if cursor:
+            cursor = ndb.Cursor(urlsafe=cursor)
+        if page < 1:
+            page = 1
+        if per_page not in [10, 20, 30, 40, 50]:
+            per_page = 30
+
+        # Sort
+        sort_field = request.args.get('sort_field', 'name')
+        sort_order = request.args.get('sort_order', 'asc')
+        if sort_field not in ['id', 'name', 'wca_person', 'roles', 'province']:
+            sort_field = 'name'
+        if sort_order.lower() not in ['asc', 'desc']:
+            sort_order = 'asc'
+
+        # Filter
+        filter_text = loads(request.args.get('filter', '')).get("q")
+
+        # Query
+        if sort_order == 'asc':
+            order_field = getattr(User, sort_field)
+        else:
+            order_field = -getattr(User, sort_field)
+        if filter_text:
+            text = filter_text.lower()
+            limit = text[:-1] + chr(ord(text[-1]) + 1)
+            users_to_show = User.query(
+                ndb.OR(
+                    ndb.AND(
+                        User.name_lower >= text,
+                        User.name_lower < limit,
+                    ),
+                    User.wca_person == ndb.Key(Person, filter_text)
+                )).order("name_lower", order_field).fetch(per_page)
+            has_more = False
+        else:
+            users_to_show, cursor, has_more = User.query(order_by=[order_field]).fetch_page(per_page,
+                                                                                            start_cursor=cursor)
+        return jsonify({
+            'data': [user.toJson() for user in users_to_show],
+            'cursor': cursor.urlsafe().decode() if cursor else '',
+            'pageInfo': {
+                'hasPreviousPage': page > 1,
+                'hasNextPage': has_more,
+            }
+        })
+
+
+@bp.route('/get_users_by_id')
+def get_users_by_id():
+    with client.context():
+        me = auth.user()
+        if not me or not me.HasAnyRole(Roles.AdminRoles()):
+            return jsonify({"error": "Forbidden"}), 403
+
+        user_ids = request.args.get('ids', "[]", type=str).strip("[]").split(",")
```

Comment:
Can't we just default to empty string on the `.get` and avoid needing the `.strip`?

Reply:
By default react admin sends the user IDs has an array, I could change that in the dataProvider and change it here but it would probably be more code than now and I don't think it makes a real improvement.

Reply:
Oh I see. I thought the `.strip` was only being used for the default case of the getter. Thank you for the clarification.